File: python/binary_search.py
# Binary search
# Sort the input
# Iniitalize two pointers for the range of the search
# Half the range - if > or < middle value

def binary_search(target, arr):
    arr.sort()
    indices = [0, len(arr)-1]
    while indices[1] - indices[0] > 0:
        middle_idx = ((indices[1] - indices[0]) // 2) + indices[0]
        middle = arr[middle_idx]
        if middle == target:
            return middle_idx
        elif target < middle:
            indices[1] = middle_idx - 1
        else:
            indices[0] = middle_idx + 1
    if arr[indices[0]] == target: return indices[0]
    return -1

File: js/binarySearch.js
exports.binarySearch = function(target, arr){
  arr.sort();
  let indices = [0, arr.length -1]
  while (indices[1] - indices[0] > 0) {
    let middleI = Math.floor((indices[1] - indices[0])/ 2) + indices[0]
    let middle = arr[middleI]
    if (middle === target) {
      return middleI
    } else if (middle > target) {
      indices[1] = middleI - 1
    } else {
      indices[0] = middleI + 1
    }
  }
  if (arr[indices[0]] === target) {
    return indices[0]
  } else {
    return -1
  }
}


